lib/composer_element_template.ex

```diff
@@ -0,0 +1,127 @@
+        %{state | frames_queues: updated_frames_queues}
+      pad == :second_input ->
+        updated_second_queue = state.frames_queues.second_queue
+        updated_second_queue = :queue.in(buffer, updated_second_queue)
+        updated_frames_queues = state.frames_queues
+        updated_frames_queues = %{updated_frames_queues | second_queue: updated_second_queue}
+        %{state | frames_queues: updated_frames_queues}
+    end
+
+    case {:queue.out(state.frames_queues.first_queue), :queue.out(state.frames_queues.second_queue)} do
+      {{{:value, first_frame_buffer}, rest_of_first_queue}, {{:value, second_frame_buffer}, rest_of_second_queue}} ->  # checks if both queues are not empty
+        {:ok, merged_frame_binary} = merge_frames(first_frame_buffer.payload, second_frame_buffer.payload)
+        merged_image_buffer = %Buffer{first_frame_buffer | payload: merged_frame_binary}  # packs merged binary into buffer (for encoder)
+        frames_queues = %{first_queue: rest_of_first_queue, second_queue: rest_of_second_queue}  # updates queues (removes processed frames)
+        state = %{state | frames_queues: frames_queues}  # updates state with reduced queues
+        {{:ok, buffer: {:output, merged_image_buffer}}, state}  # sends buffer to encoder
+      _ ->  # one of queues is empty
+        {:ok, state}
+    end
+  end
+
+  @impl true
+  def handle_caps(:first_input, %RawVideo{width: 1280, height: 720} = caps, _context, state) do
+    caps = %{caps | height: caps.height * 2}
+    {{:ok, caps: {:output, caps}}, state}
+  end
+
+  @impl true
+  def handle_caps(:second_input, %RawVideo{width: 1280, height: 720} = caps, _context, state) do
+    caps = %{caps | height: caps.height * 2}
+    {{:ok, caps: {:output, caps}}, state}
+  end
+
+
+  @impl true
+  def handle_end_of_stream(pad, _context, state) do
+    state = cond do
+      pad == :first_input ->
+        updated_frames_queues = state.frames_queues
+        updated_frames_queues = %{updated_frames_queues | first_queue: :end_of_the_stream}
+        %{state | frames_queues: updated_frames_queues}
+      pad == :second_input ->
+        updated_frames_queues = state.frames_queues
+        updated_frames_queues = %{updated_frames_queues | second_queue: :end_of_the_stream}
+        %{state | frames_queues: updated_frames_queues}
+    end
+
+    case {state.frames_queues.first_queue, state.frames_queues.second_queue} do
+      {:end_of_the_stream, :end_of_the_stream} ->  # when both stream ended
+        {{:ok, end_of_stream: :output, notify: {:end_of_stream, pad}}, state}
+      _ ->  # when only one of the streams ended
+        {:ok, state}
+    end
+  end
+
+  defp merge_frames(first_frame_binary, second_frame_binary) do
+    # TO DO: implement function, that merge two raw video, 1280x720, I420 encoded frames,
+    # into raw video, 1280x720, I420 encoded frame
+    {:ok, merged_frame_binary}
+  end
```

Comment:
Instead of this, take `:implementation` as one of the [options](https://hexdocs.pm/membrane_core/Membrane.Element.Base.html#def_options/1) of an element, maybe of type `:ffmpeg | :nx | :opengl`. Then, keep in the state the `compositor_module` based on this option, so something like `Membrane.VideoCompositor.Nx` for `:nx`, etc. Then, just call `compositor_module.merge_frames(...)` in `handle_process`.

Reply:
Also, you can add Nx module (maybe just a mock) so it would compile nicely and pass in circleci